web/client/reducers/geostory.js

```diff
@@ -208,6 +226,26 @@ export default (state = INITIAL_STATE, action) => {
     case TOGGLE_CARD_PREVIEW: {
         return set('isCollapsed', !state.isCollapsed, state);
     }
+    case TOGGLE_SETTINGS: {
+        const visibility = get(state, `currentStory.settings.${action.option}`);
+        return set(`currentStory.settings.${action.option}`, !visibility, state);
+    }
+    case TOGGLE_SETTINGS_PANEL: {
+
+        // when opening, the expanded items will be calculated based on checked ones
+        const newStatus = !state.isSettingsEnabled;
```

Comment:
You shouldn't use selectors in reducers ( I see `selectedCardSelector` is already used, we should check if we can remove that too ) because: 
 - they may cache results and have unwanted side-effects
 - actually they are written at application scope (geostory.<state>) while the reducer doesn't know it's name. We should evaluate before to do this.
Expanded is a derived information from selected. therefore please create instead a selector outside the reducer to derive expanded from checked instead of saving this duplicated information in the state.

Reply:
- i will remove the selectors form reducers
- expanded, i would suggest to edit the settingsSelector to add expanded info, because actually they are present in settings